cmd/kube-apiserver/app/server.go

```diff
@@ -564,13 +564,19 @@ func BuildAuthenticator(s *options.ServerRunOptions, storageFactory serverstorag
 		if err != nil {
 			return nil, nil, fmt.Errorf("unable to get serviceaccounts storage: %v", err)
 		}
+		storageConfigPods, err := storageFactory.NewConfig(api.Resource("pods"))
```

Comment:
I'm pretty sure the above comment ("clientsets fail when they're initialized with some API versions excluded") is obsolete. a follow up to switch everything to `NewGetterFromClient()` and remove NewGetterFromStorageInterface() completely would probably be good.

Reply:
actually, would it make sense to switch to a client now, rather than adding a pod registry?